### DISCUSSION TITLE
TileSet Terrain slope and multiple terrains per tile

### DISCUSSION THREAD

**Discussion by hsandt** [godotengine/godot-proposals#6157] [Editor]
Building side-view levels with terrains is much more difficult that top-view levels when you start adding slopes.

I found a tutorial for Godot 3: https://www.youtube.com/watch?v=dCL687AQbHM that exploits the fact that you can create multiple Auto-tiles on the same tileset, to create one version with common tiles + flat ground, another one with common tiles + slopes (repeat for every slope angle), and adding a script to detect neighboring tiles from another auto-tileset, it manages to join all the tiles together.

However, in Godot 4, you can only define one Terrain property per tile, so you cannot use this trick unless you copy the full spritesheet (or split it in two versions, one with slopes and one without).

I didn't try the neighbor tile detection script yet, but if it works in Godot 4 then at least we'd be able to join slope tiles to the rest of the tilemap, filling the gaps with the correct tile sprites (we'd still need sprite duplication so it works both ways).

How do you handle slopes in side-view games, and did you manage to make it work with auto-tiles?

Currently, I'm just adding slope tiles manually and the "flat ground terrain" doesn't fill the gaps properly (may be helped with the mentioned tilemap script):

![image](https://user-images.githubusercontent.com/1067772/214130212-da34b6ef-69b2-4016-b245-e0e0027cc534.png)

Related issue: https://github.com/godotengine/godot-proposals/issues/3200

**Comment by hsandt** [comment]
So, I tried to use the script technique from the video at https://youtu.be/dCL687AQbHM?t=363, but TileSet doesn't have an is_tile_bound method anymore since the departure from the old AutoTile system. So I'm not sure how to recognize tiles from a different terrain as neighbors.

**Comment by dandeliondino** [comment]
I saw another proposal mention being able to assign multiple terrains to a single bit, which should achieve this. It could be tedious depending on how it is implemented in the UI, but maybe you could define a terrain groups and paint it as you would a single terrain?

I would also love to see some of the matching algorithm exposed to scripting (like is_tile_bound was in 3).
